test/integration/test_groundlight.py

```diff
@@ -320,6 +320,53 @@ def test_submit_image_query_png(gl: Groundlight, detector: Detector):
     assert is_valid_display_result(_image_query.result)
 
 
+def test_submit_image_query_with_id(gl: Groundlight, detector: Detector):
+    # submit_image_query
+    id = f"iq_{KsuidMs()}"
+    _image_query = gl.submit_image_query(
+        detector=detector.id, image="test/assets/dog.jpeg", wait=10, human_review="NEVER", image_query_id=id
+    )
+    assert str(_image_query)
+    assert isinstance(_image_query, ImageQuery)
+    assert is_valid_display_result(_image_query.result)
+    assert _image_query.metadata is not None
+    assert _image_query.metadata.get("is_from_edge")
```

Comment:
Why not check that the returned ID is actually the one you submitted?

Reply:
Good point, it definitely should check that!